### DISCUSSION TITLE
Unable to repair wsl distribution when using WSL1

### DISCUSSION THREAD

**Discussion by tmprnc** [Difegue/LANraragi#952] [Q&A]
**LRR Version and OS**
LANraragi 0.9.0, reinstalled, with Karen.exe.config having been deleted from the folder.
Windows 11 Pro (build 10.0.22635.3209), running native (no Docker)

**Bug Details**
LRR is unable to repair the wsl1 distro. Everytime it attempts to run wsl.exe, it crashes and returns exit code -1. It's _probably_ a WSL problem and not necessarily LRR itself. WSL2 works, but I can't use WSL2 because it automatically turns on Hyper-V, and I also happen to run an Android emulator for games which runs slow to an unplayable extent when it is turned on.
Are there steps I can take to completely clean out the distro? Would I get a better shot by just going through the trouble of installing Docker?

**Matching Logs**
Post any logs you have encountered when reproducing the bug.  
If possible, switch the app to Debug Mode in order to have more logs.

**Screenshots**
![1708440028](https://github.com/Difegue/LANraragi/assets/3785923/86026e51-8fcb-41db-a474-1cef2e669c2b)
![1708440158](https://github.com/Difegue/LANraragi/assets/3785923/a2704ecb-23f6-41ee-a757-fe87b4c06f56)

**Comment by Difegue** [comment]
Have you tried running those "using wsl CLI" commands in a terminal to see what the error messages actually are? That might give you some pointers.  

As a heads-up, WSL1 is just being left to die by MS at this point so it's likely the windows version will switch to WSL2-only in future releases.

**Reply by tmprnc** [reply]
WSL was giving off an error because it was missing a feature. Using the command to repair it, I was able to repair the distro.
However, I run into two problems.
1. Running a new instance of the LANraragi GUI tells me that it's already running, despite being nowhere in the system tray.
![1708443495](https://github.com/Difegue/LANraragi/assets/3785923/d4b968f1-d93c-40de-aacf-fa7941c37fbb)
2. Once more, something that upsets the emulator is turned on. It claims it as Hyper-V (that is not the case, it is disabled in Windows features), and I think it is the Virtual Machine Platform. Once more, the emulator runs sluggish.
![1708443410](https://github.com/Difegue/LANraragi/assets/3785923/e0dbc369-8f8b-49ad-a55f-8e84fa7d1924)

**Reply by Difegue** [reply]
WSL2 will inevitably use the windows VM platform if it's in use; I unfortunately don't really have any workarounds for this except switching to Linux.. Docker on Windows will also rely on WSL2.
